barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -844,85 +802,81 @@ typename cycle_group<Builder>::batch_mul_internal_output cycle_group<Builder>::_
         operation_hints.emplace_back(element.x, element.y);
     }
 
-    // Construct an in-circuit straus lookup table for each point
+    // Construct in-circuit Straus lookup tables for each point.
+    // ROM tables enable efficient scalar multiplication by replacing doublings with table lookups.
     std::vector<straus_lookup_table> point_tables;
-    const size_t hints_per_table = (1ULL << TABLE_BITS) - 1;
+    point_tables.reserve(num_points);
+    const size_t hints_per_table = (1ULL << ROM_TABLE_BITS) - 1;
     OriginTag tag{};
     for (size_t i = 0; i < num_points; ++i) {
         // Merge tags
         tag = OriginTag(tag, scalars[i].get_origin_tag(), base_points[i].get_origin_tag());
 
+        // Construct Straus table
         std::span<AffineElement> table_hints(&operation_hints[i * hints_per_table], hints_per_table);
-        point_tables.emplace_back(context, base_points[i], offset_generators[i + 1], TABLE_BITS, table_hints);
+        straus_lookup_table table(context, base_points[i], offset_generators[i + 1], ROM_TABLE_BITS, table_hints);
+        point_tables.push_back(std::move(table));
     }
 
+    // Initialize pointer to traverse the precomputed Straus algorithm hints (not the table construction hints)
     AffineElement* hint_ptr = &operation_hints[num_points * hints_per_table];
     cycle_group accumulator = offset_generators[0];
 
-    // populate the set of points we are going to add into our accumulator, *before* we do any ECC operations
-    // this way we are able to fuse mutliple ecc add / ecc double operations and reduce total gate count.
-    // (ecc add/ecc double gates normally cost 2 Ultra gates. However if we chain add->add, add->double,
-    // double->add, double->double, they only cost one)
-    std::vector<cycle_group> points_to_add;
```

Comment:
This is another case where we can simplify due to the fact that we now organize the circuit into gate-type blocks. This logic was previously separated with the only purpose of ensuring that the ECC addition gates were consecutive - bit this now happens by default.